### DISCUSSION TITLE
Cancel Review Environment protection rule via concurrency.cancel-in-progress

### DISCUSSION THREAD

**Discussion by Fresa** [community/community#40357] [Actions]
### Select Topic Area

Product Feedback

### Body

It would be great if it were possible to only cancel workflows in the same group that currently is in review as part of a jobs Environment protection rule! Maybe a new setting under the `concurrency` configuration, `cancel-in-review`?

We use the review environment protection rule feature to review Terraform plans before applying them, but if there are new changes before a plan has been reviewed it's quite pointless to review and apply. `cancel-in-progress` [seems to not gracefully cancel](https://github.com/orgs/community/discussions/26311) running jobs, and since cancelling terraform plan and apply abruptly can cause both state corruption as corrupted/inconsistent provisioned resources, this could be an alternative to solving that.

**Comment by ossanna16** [comment]
Hi there @Fresa and welcome to our community! Thank you for the feedback!
